packages/data-stores/src/site/actions.ts

```diff
@@ -423,9 +418,7 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 			yield* setGlobalStyles( siteSlug, activatedTheme.stylesheet, globalStyles, activatedTheme );
 		}
 
-		if ( shouldRunThemeSetup ) {
-			yield* runThemeSetupOnSite( siteSlug );
-		}
+		yield* runThemeSetupOnSite( siteSlug );
```

Comment:
![badge](https://img.shields.io/badge/review-nits-green.svg)

I thought it'd be nice to have a comment about when exactly we need to run `/theme-setup`. (Or can we even prevent that call?)

Reply:
We will always call this endpoint. Could you elaborate more?

Reply:
I assume we only need to insert posts when the homepage template contains a Query block. So, it may feel a bit strange that "theme setup" is executed in any case, since we cannot know what will happen by calling this function at first glance. (![badge](https://img.shields.io/badge/review-ask-blue.svg) Like, when the homepage template does NOT contain a Query block, there are no other side effects by this change?)

Maybe the following pseudo code feels more intuitive. 

```js
yield* maybePopulatePosts() // We can imagine this will populate posts in a specific condition and do nothing else
```

or

```js
if (homepage_has_blog_patterns) {
  yield* populatePosts()
}
```

Does it make sense?
<!--
drafts

So, when the homepage template does not contain a Query block, I thought we don't need to call this endpoint (`/theme-setup`) in the first place. 
I was wondering what's the side effect when we always run `/theme-setup`.

Because we don't need to insert post
In that case, I thought the logic in `homepage_has_blog_patterns()` (in D121838-code) could be used not to call the request. -->

Reply:
The problem is, we can only check if the homepage contains blog patterns (or at least it is way simpler to do) in the backend :') 

> Like, when the homepage template does NOT contain a Query block, there are no other side effects by this change?

Correct.

I'll try to add some comments to make it clear.

Reply:
Added some comments 🙈